core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -1298,18 +1298,18 @@ class SizeBasedCoalescer(val maxSize: Int) extends PartitionCoalescer with Seria
       val splitSize = fileSplit.getLength
       if (currentSum + splitSize < maxSize) {
         addPartition(partition, splitSize)
-        index += 1
-        if (index == partitions.size) {
-          updateGroups
-        }
       } else {
-        if (currentGroup.partitions.size == 0) {
+        if (currentGroup.partitions.isEmpty) {
           addPartition(partition, splitSize)
-          index += 1
         } else {
-          updateGroups
```

Comment:
can you point out which line/lines cause the bug?

Reply:
> can you point out which line/lines cause the bug?

In the pr description I have show the case this bug happen. Not one line or some lines cause this error.

Reply:
> can you point out which line/lines cause the bug?

Change the desc make it more clear.